Refactor the compose pipeline

edilmedeiros opened the pull request.

edilmedeiros commented:
When reviewing, please don't forget to check if the readme commands work properly.

m4ycon commented on node.signet.yml:
```diff
@@ -0,0 +1,47 @@
+    environment:
+      BITCOIN_NETWORK: signet
+    ports:
+      - "38332:38332"
+      - "38333:38333"
```
does it make sense if we define the "outside" port to be a specific one, always? Like there would be no diff between regtest, signet or mainnet from outside the container.

edilmedeiros replied:
Then we would need to include a `bitcoin.conf` config file in the image configuring the node to use custom ports. I probably prefer to use the standard ports for the instrumentation node so that I can use `bitcoin-cli` on the host to drive it without the need to know about the custom configuration.

But I didn't test it yet.

m4ycon commented on README.md:
```diff
@@ -102,13 +104,13 @@ Bitcoin blockchain data is stored in the `bitcoin-data` Docker volume. This ensu
 
 # Build NATS
-docker-compose build nats
+   docker compose -f node.regtest.yml build nats
```
```suggestion
docker compose -f node.regtest.yml build nats
```

edilmedeiros commented on node.signet.yml:
```diff
@@ -0,0 +1,47 @@
+      retries: 3
+      start_period: 5s
+
+  nats:
+    build:
+      context: .
+      dockerfile: docker/nats.dockerfile
+    ports:
+      - "4222:4222"
+    healthcheck:
+      test: ["CMD", "wget", "-q", "--spider", "http://localhost:8222/healthz"]
+      interval: 1m
+      timeout: 1s
+      retries: 3
+      start_period: 1s
```
Let me take a look at these docs and play with them.

edilmedeiros commented:
@m4ycon thanks for the suggestion of using includes, I think it removed a lot of duplicate code. Before merging, let me check also the profile feature, looks like it can simplify adding the tools to the pipeline without complicating the command line.

m4ycon commented on README.md:
````diff
@@ -23,53 +29,49 @@ The system consists of several interconnected services:
+2. **Build and start basic system with the `logger` tool**:
    ```bash
-   docker-compose up -d
+   docker compose -f node.regtest.yml -f docker-compose/logger.yml up -d
````
I also miss something about the profiles you created, here in readme, btw looks like a very cool feat.

```bash
docker compose --profile monitoring up
```

edilmedeiros replied:
Yeah, didn't rewrite the docs yet.

Now we can call like `docker compose --profile logger up`; add and remove running tools without getting everything down.

edilmedeiros commented on node.signet.yml:
```diff
@@ -0,0 +1,8 @@
+  - path: docker-compose/nats.yml
+  - path: docker-compose/node.usdt.yml
+  
+volumes:
+  bitcoin-data:
+  bitcoin-bin:
```
I'll test it.

edilmedeiros commented:
1. Renamed the tools services excluding the `peer-` prefix.
2. Updated documentation.
3. Rebased on top of master, preparing to merge.

I think this is good to merge after some else tests it, preferably checking that the docs make sense and that the provided commands work.

Let's do the volume refactor in a follow-up. I want to focus on other stuff now: adding another regtest container that creates some traffic for testing and development.

edilmedeiros commented:
There's still a bug because I renamed the files. I hope this is the last time I have to rebase, this is becoming quite annoying...

m4ycon reviewed (approved):
seems good!

Just commited a little fix on docker profiles removing "peer" preffix, must have gone unnoticed.
